### ISSUE TITLE
code ethwallpaper website

### ISSUE THREAD

**Issue by owocki** [gitcoinco/skunkworks#5]
Requirements

# Upload

1. Take the winning design from https://github.com/gitcoinco/skunkworks/issues/4
2. and the code from https://github.com/gitcoinco/skunkworks/pull/3
3. create a website that can accept image uploads, validate them, and then spit out the resultant 'eth wallpaper' image. 
4. after image is uploaded, it should be displayed to the user with an option to tweet it.

# Gallery

1. any generated ethwallpaper image should go into the gallery.
2. two dynamic pieces of functionality should exist on each gallery image.  'like' and 'report'
3. any offensive images can/should be removed, if an image has more 'report's than 'likes' it should be hidden from the gallery

# Other Reqquirements

I'd prefer python/django be used for this, but it's up to the submitter at the end of the day.  Choose the right tool for the job.

# Bounty

1. To claim this bounty, please comment below and claim it on gitcoin.co
2. Please do not start work until you've claimed the bounty.

**Comment by gitcoinbot** [comment]
__This issue now has a funding of 0.49 ETH (284.2 USD) attached to it.__

 * If you would like to work on this issue you can claim it [here](https://gitcoin.co/funding/details?url=https://github.com/gitcoinco/skunkworks/issues/5).
 * If you've completed this issue and want to claim the bounty you can do so [here](https://gitcoin.co/funding/details?url=https://github.com/gitcoinco/skunkworks/issues/5)

**Comment by tra38** [comment]
This sounds like a very interesting project, and I would like to take it. Claiming it now...

**Comment by owocki** [comment]
@tra38 see https://github.com/gitcoinco/skunkworks/pull/6 -- might require a lot of CSS mastery :)

> EDIT: When I tried claiming it, I got a message stating "Could not get bounty details."


thanks for the bug report.  im waiting on this issue to be resolved to fix this https://github.com/MetaMask/metamask-extension/issues/2736

**Comment by tra38** [comment]
@owocki: Yeah, maybe I have bitten off more than I chewed. :) I'll back off on that issue for now. I'm very confident in back-end work though.

**Comment by owocki** [comment]
word; maybe you can tag team it with someone who does frontend stuff :)

**Comment by merryt** [comment]
@tra38 I could help out if you need a hand, are you on the gitcoin slack channel? shoot me a DM and we will talk about how we can split up work.

**Comment by oogetyboogety** [comment]
Would be interested in claiming this issue to begin working on it with Python/Django.

**Comment by thelostone-mc** [comment]
I have the bandwidth to help out ^_^ Leave me a ping if you need an extra hand 
@tra38 @merryt

**Comment by stojce** [comment]
I tried to make a claim, but nothing happens after I went through claim process on Gitcoin (except that some ETH was withdrawn for gas...). 
@owocki is Metamask extension fixed? I would love to work on this if someone elase is not working already.

**Comment by owocki** [comment]
im still seeing intermittent infura / metamask issues.  am waiting to hear back from them.

@tra38 @thelostone-mc @oogetyboogety @merryt @stojce ill leave it up to you all to figure out who is gonna work on this.  last i spoke to @tra38 and @merryt it looked like they were about to get started soon

**Comment by gitcoinbot** [comment]
__The funding of 0.49 ETH (330.75 USD) attached has been claimed by @stojce.__ 

 @stojce, please leave a comment to let the funder (@owocki) and the other parties involved your implementation plan.  If you don't leave a comment, the funder may expire your claim at their discretion. 

 * Learn more at: https://gitcoin.co/funding/details?url=https://github.com/gitcoinco/skunkworks/issues/5
 * Questions? Get help on the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $9757.26 more Funded OSS Work Available at: https://gitcoin.co/explorer

**Comment by owocki** [comment]
and... like clockwork.  theres your claim. @stojce 

i just invited everyone to a priv channel on the gitcoin slack to coordinate.  id love to see yall work togehter.

@oogetyboogety i couldnt find you on gitcoin.co/slack.  ping me (im owocki) if you want in

**Comment by owocki** [comment]
@stojce you wanna take the lead on this!? @tra38 is out :P

**Comment by stojce** [comment]
@owocki Made PR at https://github.com/gitcoinco/skunkworks/pull/7
Let me know if you need any help testing this.

**Comment by owocki** [comment]
@stojce 

this is awesome!

notes on http://ethwallpaper.co/

1. the ethereum logo should be white, not black (up top)
2. you can link 'who are we' to gitcoin.co/about?source=ethwallpaper' for now
3. i get a 500 when i try to upload though http://bits.owocki.com/0G2I45231T0k/Screen%20Shot%202017-12-21%20at%202.38.32%20PM.png
4. since "generator script runs pretty slow - it takes 6-7 min to create wallpaper"  maybe well have to add a queueing mechanism and email notification or something…
5. do we need 'title', 'author', 'size' and 'tags' here? http://bits.owocki.com/3R0X0u0r1V0S/Screen%20Shot%202017-12-21%20at%202.42.40%20PM.png
6. does the heart button work? it just seems to take you to the image speciifc page
7. can we change 'from colorado community' to 'from gitcoin community'. its more accurate, and i want to give you all credit!
8. i think we need some 'report' mechanism for if someone uploads something offensive. this was in the original scope "any offensive images can/should be removed, if an image has more 'report's than 'likes' it should be hidden from the gallery"
9. not sure if this works 'after image is uploaded, it should be displayed to the user with an option to tweet it.' since i can't seem to upload
10.  

@stojce sorry to overhwhelm you with feedback.  the project looks like its really coming along!  i'm happy to provision a v2 bounty to grease the wheels if any of this feels out of scope / overhwleming :)

**Comment by stojce** [comment]
@owocki

1. Please provide graphical assets;
2. Will do;
3. This is already fixed;
4. We should discuss bit more about this, maybe even as a new task; '
5. Yep, this is by design. Do you have different idea in mind?
6. It works, but  counter refresh doesn't - will fix this;
7. Sure;
8. Report mechanism is there, but not implemented on front-end. Where do
you need that button / icon?
9. Please provide the text of the tweet;

**Comment by owocki** [comment]
> 1. Please provide graphical assets;

looks like @merryt provided the assets in his PR.  https://github.com/gitcoinco/skunkworks/issues/4#issuecomment-351216280

you could probably get by with a `filter: invert(100%);` in CSS though.

> 4. We should discuss bit more about this, maybe even as a new task; '

Yes, this is certainly new scope!

> 5. Yep, this is by design. Do you have different idea in mind?

My first thought was that it's kind of annoying to have to categorize it... but i suppose not a big deal for MVP.

> 8. Report mechanism is there, but not implemented on front-end. Where do you need that button / icon?

Right next to the 'heart' button is fine.

> 9. Please provide the text of the tweet;

Check out this #ETH Wallpaper: <link>

Thanks!  I'm Excited about this.

**Comment by stojce** [comment]
@owocki everything is fixed and deployed.

**Comment by stojce** [comment]
Thoughts on "3. slow script": I think using webscokets would be best - just show small progress notification on top left corner...

**Comment by owocki** [comment]
> Thoughts on "3. slow script": I think using webscokets would be best - just show small progress notification on top left corner...

seems like a good idea to me

**Comment by owocki** [comment]
more testing notes: search should be case insensitive

**Comment by owocki** [comment]
more testing notes: google analytics

**Comment by owocki** [comment]
watermark in bottom right: ethwallpaper.co

**Comment by owocki** [comment]
idea: charge 0.01 ETH per wallpaper generated

**Comment by owocki** [comment]
idea: more filters a la https://twitter.com/rabbyte/status/944676368641433600

**Comment by gitcoinbot** [comment]
__The funding of 0.49 ETH (351.79 USD) attached to this issue has been approved & issued to @stojce.__  

 * Learn more at [on the gitcoin issue page](https://gitcoin.co/funding/details?url=https://github.com/gitcoinco/skunkworks/issues/5)
 * Questions? Get help on the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $10912.77 more Funded OSS Work Available at: https://gitcoin.co/explorer
